client.c

```diff
@@ -1362,6 +1365,26 @@ ssize_t read(int fd, void *buf, size_t count) {
 		}
 
 		return count;
+	} else {
+		int t = get_timer_from_fd(fd);
+		if (t >= 0) {
+			if (count >= sizeof(timers[t].expired)) {
+				do {
+					if (timers[t].expired) {
+						memcpy(buf, &timers[t].expired, sizeof(timers[t].expired));
+						timers[t].expired = 0;
+						return sizeof(timers[t].expired);
+					} else if (timers[t].flags & TFD_NONBLOCK) {
+						errno = EAGAIN;
+						return -1;
+					}
+					usleep(10);
```

Comment:
I guess this should be a call of `select`, but if we don't have an application that would make blocking reads, I think it's better to fail an assertion.

Reply:
I'm sorry. I don't understand your comment. Can you provide an example?
Read is fine on both blocking and non blocking fd timers.

Reply:
`usleep(10)` doesn't block for the right amount of time. It needs to translated to a `select` or `poll` call. If you don't have any application which would do blocking reads, I'd prefer to leave this unimplemented.

Reply:
is assert(0); instead usleep() enough for you?

Reply:
`assert(timers[t].fd_flags & TFD_NONBLOCK) ;` at the beginning of the branch would look better.

Reply:
I done a little bit different. Only for linuxptp testing purpose we should never read before a timer expiration.